ckanext/opendatani/helpers.py

```diff
@@ -105,3 +112,31 @@ def activity_list_to_text(activity_stream):
                               'is_new': activity.get('is_new', False),
                               'dataset_url': get_dataset_url(activity)})
     return activity_list
+
+
+def get_storage_path_for(dirname):
+
+    """Returns the full path for the specified directory name within
+    CKAN's storage path. If the target directory does not exists, it
+    gets created.
+    :param dirname: the directory name
+    :type dirname: string
+    :returns: a full path for the specified directory name within CKAN's storage path
+    :rtype: string
+    """
+
+    storage_path = config.get('ckan.storage_path')
+    target_path = os.path.join(storage_path, 'storage', dirname)
+
+    if not os.path.exists(target_path):
+        try:
+            os.makedirs(target_path)
+        except OSError as exc:
+            log.error('Storage directory creation failed. Error: %s' % exc)
+            target_path = os.path.join(storage_path, 'storage')
+
+            if not os.path.exists(target_path):
+                log.info('CKAN storage directory not found also')
+                raise
+
+    return target_path
```

Comment:
This is nice but I would recommend using  `/tmp` directory for this purpose, we don't want to keep those files, right? all we need is to create a dummy file for a download purpose only and delete after... This will also work but think it complicates stuff..